### PR TITLE
feature(show): support jsonpath output for `helm show value`

### LINKED ISSUE #7787: `helm show values` should have a way to specify what values to show
`Helm show values` doesn't have a way to specify which values you want to output. It would be nice to have this because  tools like bash don't have a yaml parser natively so to get a specific value from a chart I have to download a yaml parser. I believe helm should have this integrated because helm seems to be using something like this for `helm install --set` already.

### PR DESCRIPTION
**Pull request by donggangcj**
**What this PR does / why we need it**:

Closes #7787

**Special notes for your reviewer**:
1. This is a preliminary realization. I will improve the implementation after listening to everyone's advice.
2. Only `helm show values` support the jsonpath format.
3. This is  implemented through Library `k8s.io/cli-runtime` and related document is [here](https://kubernetes.io/docs/reference/kubectl/jsonpath/).

**If applicable**:
- [ ] this PR contains documentation
- [x] this PR contains unit tests
- [ ] this PR has been tested for backwards compatibility

### CODE REVIEW

File: cmd/helm/show.go [hunk 1]
```diff
@@ -151,6 +151,9 @@ func addShowFlags(subCmd *cobra.Command, client *action.Show) {
 	f := subCmd.Flags()
 
 	f.BoolVar(&client.Devel, "devel", false, "use development versions, too. Equivalent to version '>0.0.0-0'. If --version is set, this is ignored")
+	if subCmd.Name() == "values" {
+		f.StringVar(&client.JSONPathTemplate, "jsonpath", "", "use jsonpath output format")
```
Comment (mattfarina) [inline cmd/helm/show.go:155]: The help text here isn't very descriptive. A user unfamiliar with JSONPath might think this is a binary flag. Could the help text read something like:

> supply a JSONPath expression to filter the output

**Review by mattfarina — COMMENTED** [review]
This mostly looks good. I can imagine using this flag and pattern elsewhere. I have one nit about the language of the flag. Could you please update it.

File: cmd/helm/show.go (hunk 1 above)
Comment (donggangcj) [reply cmd/helm/show.go]: ok! done;

**Review by jdolitsky — APPROVED** [review]
LGTM

File: pkg/action/show.go [hunk 2]
```diff
@@ -87,9 +89,17 @@ func (s *Show) Run(chartpath string) (string, error) {
 		if s.OutputFormat == ShowAll {
 			fmt.Fprintln(&out, "---")
 		}
-		for _, f := range s.chart.Raw {
-			if f.Name == chartutil.ValuesfileName {
-				fmt.Fprintln(&out, string(f.Data))
+		if s.JSONPathTemplate != "" {
+			printer, err := printers.NewJSONPathPrinter(s.JSONPathTemplate)
+			if err != nil {
+				return "", fmt.Errorf("error parsing jsonpath %s, %v", s.JSONPathTemplate, err)
```
Comment (hickeyma) [inline pkg/action/show.go:95]: Better to use `errors.Wrapf`. Pattern used throughout codebase including https://github.com/helm/helm/blob/master/pkg/action/install.go#L144.

**Review by hickeyma — CHANGES_REQUESTED** [review]
Thanks for pushing this functionality @donggangcj. I think it will be useful to users.

Small change inline.

File: pkg/action/show.go (hunk 2 above)
Comment (donggangcj) [reply pkg/action/show.go]: Thank you for your detailed instructions！ Done;

**Review by hickeyma — APPROVED** [review]
LGTM, thanks @donggangcj

**Pull request merged**